src/vault.rs

```diff
@@ -760,3 +779,9 @@ enum EventResult {
     // `RestartRequired` event received.
     Restart,
 }
+
+#[derive(Deserialize, Serialize, PartialEq, Eq, Debug, Clone)]
+pub enum Refresh {
```

Comment:
Both `maid_manager` and `data_manager` modules now define their own refresh types, so I'd prefer this to be changed to:
```rust
pub enum Refresh {
    MaidManager(maid_manager::Refresh),
    DataManager(data_manager::Refresh),
}
```
so we could avoid the double serialization.